src/shared/Microsoft.Git.CredentialManager/Interop/Posix/PosixEnvironment.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Git.CredentialManager.Interop.Posix
+{
+    public abstract class PosixEnvironment : IEnvironment
+    {
+        public PosixEnvironment()
+        {
+            var dict = new Dictionary<string, string>();
+            var variables = Environment.GetEnvironmentVariables();
+
+            foreach (var key in variables.Keys)
+            {
+                if (key is string name && variables[key] is string value)
+                {
+                    dict[name] = value;
+                }
+            }
+
+            Variables = dict;
+        }
+
+        public IReadOnlyDictionary<string, string> Variables { get; }
+
+        public bool IsDirectoryOnPath(string directoryPath)
+        {
+            if (Variables.TryGetValue("PATH", out string pathValue))
+            {
+                string[] paths = pathValue.Split(':');
+                return paths.Any(x => IsSamePath(x, directoryPath));
+            }
+
+            return false;
+        }
+
+        public virtual void AddDirectoryToPath(string directoryPath, bool system)
+        {
+            throw new NotImplementedException();
```

Comment:
We don't bother implementing the $PATH manipulation methods for POSIX (or specifically for macOS) because we don't need them (yet).

The postinstall script run by the package installer creates a GCM Core symlink in /usr/local/bin which is already on the $PATH.